libdrgn/python/thread.c

```diff
@@ -0,0 +1,122 @@
+#include "drgnpy.h"
+#include "helpers.h"
+
+static void Thread_dealloc(Thread *self)
+{
+	Py_DECREF(self->prog);
```

Comment:
We usually use `Py_XDECREF` in `dealloc` in case we end up destroying an object that wasn't fully initialized:

```suggestion
	Py_XDECREF(self->prog);
```

This doesn't matter for the existing uses of `Thread`, but it's easy to miss if we end up creating these objects in a different codepath.